File: matlab_code/oldcode/shiyan1_4.m
A=imread("trees.tif");
A=im2double(A);
[m,n]=size(A);
d1=ones(1,m)*2;
d2=ones(1,m-1)*(-1);
d1=diag(d1);
d2=diag(d2,1);
k=d1+d2+d2';
d2A=k*A;
d2A_2=0;
for i =1:m
    d2A_2=d2A_2+k(:,i)*A(i,:);
end
subplot(2,2,1);imagesc(A);
subplot(2,2,2);imagesc(d2A);
subplot(2,2,3);imagesc(d2A_2);

File: anacondacode/tensorflow_environment/old_code/计算方法/shiyan1_4.py
# -*- coding: utf-8 -*-
"""
Created on Fri Nov 22 11:27:39 2019

@author: LJZ
"""

import numpy as np
import matplotlib.pyplot as plt
from skimage import data
A=data.coins().astype(np.float64)
#plt.imshow(A)
[m,n]=np.shape(A)
d1=np.ones(m)*2
d2=np.ones(m-1)*(-1)
d1=np.diag(d1)
d2=np.diag(d2,1)
k=d1+d2+d2.T
d2A=np.matmul(k,A)
fig=plt.figure()
plt.subplot(2,2,1);plt.imshow(A)
plt.subplot(2,2,2);plt.imshow(d2A)
print("用外积法实现k*A")
print("用图像举例，直观表示矩阵")
d2A_2=0
for i in range(m):
    d2A_2+=np.tensordot(k[:,i],A[i,:],axes=0)
plt.subplot(2,2,3);plt.imshow(d2A_2)